zipp/__init__.py

```diff
@@ -313,6 +322,47 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
+
+    def is_symlink(self):
+        return False  # See #82102
+
+    @contextlib.contextmanager
+    def _scandir(self):  # Needed for glob
+        yield self.iterdir()
+
+    def _make_child_relpath(self, relpath):  # Needed for glob
+        return self / relpath
+
+    def glob(self, pattern):
+        if not pattern:
+            raise ValueError("Unacceptable pattern: {!r}".format(pattern))
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
+            yield p
+
+    def rglob(self, pattern):
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern and pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(("**",) + tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
+            yield p
```

Comment:
This looks like a lot of repetition. Why not

```suggestion
    def rglob(self, pattern):
        return self.glob(f'**/{pattern}')
```

Reply:
I implemented this approach in e587f319826bce03df1dc7233842f054172c1fd2.